docs/metrics/index.md

```diff
@@ -1,269 +1,140 @@
-
-Datadog is an extension to [statsd](https://github.com/statsd/statsd) supporting everything in statsd and some additional metrics and 
-augmentary data on top. Reporting to Datadog can be tested by creating a free test account with Datadog and configuring an agent on the 
-developers machine used to forward captured metrics to Datadog. 
-
-The Datadog instrumentation will send its metrics directly to the Datadog agent or a statsd instance using UDP datagrams. The protocol is 
-implemented within ZMX and does not require additional dependencies. 
-
-Prometheus is an alternative to Datadog / Statsd and is also widely used to monitor application. A prometheus agent is normally configured 
-to poll metrics from various sources. These sources are also known as _exporters_ in the prometheus universe. Each exporter provides a HTTP 
-endpoint providing the current state of all collected metrics in a [text format](https://prometheus.io/docs/instrumenting/exposition_formats/)
-specified by prometheus. 
-
-Very often, prometheus is used in conjunction with [Grafana](https://grafana.com/) to display the collected metrics in dashboards.
-
-The prometheus instrumentation collects and aggregates the metrics from the metric ebent stream and provides a method to produce the report 
-according to the prometheus specification. At the moment we do not provide the HTTP endpoint in the ZMX production code, but only show an 
-example how the report can be served via a simple HTTP endpoint in the test code. 
-
-## Basics 
-
-The ZMX supported metrics are defined in the package `zio.zmx.metrics`. All metrics are captured as instances of `MetricEvent` shown below. 
-
-```scala mdoc:reset-object
+```scala mdoc:invisible
 import zio._
-import zio.zmx._
+import zio.random._
+import zio.duration._
 import zio.zmx.metrics._
-
-final case class MetricEvent(
-  name: String,
-  details: MetricEventDetails,
-  timestamp: Option[java.time.Instant],
-  tags: Chunk[Label]
-) {
-  def metricKey: String =
-    ???
-}
+import zio.zmx.state.DoubleHistogramBuckets
 ```
 
-Each event has a `name` and a potential empty chunk of `Label`s identifying the metric. A `Label` is simply a key/value pair that can be used
-to further qualify the metrics. Within the reporting backends normally an aggregation for metrics with the name can be retrieved, while 
-the labels are normally used for a further drill down. 
-
-Within ZMX `name` and `Label`s are used as a composed key to capture and report the metrics. 
-
-Finally, the event carries the actual measurement as an instance of `MetricEventDetails`. Refer to the reminder of this page to 
-learn about supported metric types.
-
-ZMX uses smart constructors to instantiate metric events. These normally have a result type of `Option[MetricEvent]`. As a consequence, 
-a metric that cannot be constructed will be swallowed silently with no further error reporting. 
+## Counter
```

Comment:
Could be good to have a brief section before this providing an overview of the different metric types that are supported and perhaps their equivalents for different backends.

Reply:
I have added an intro, but I was thinking to keep the reference free from any backends to emphasize that ZMX is backend agnostic. Prometheus and StatsD have their own dedicated pages - probably will add a link from the reference

Reply:
That's fine. I like the idea of keeping the introduction abstract from individual backends but I do think a lot of people are going to be coming from that background so could be nice to add a table showing how the different metric types map to each other add some point, maybe in another file, for those who are interested.